rsp/xxx_rest_api_design.md

```diff
@@ -0,0 +1,261 @@
+
+Parameters:
+* `type`: Restricts search to the specified type ("monster" or "player")
+
+### Returns
+
+```javascript
+[
+{
+    "id": 4,
+    "type": {
+        "player": {
+            "id": 1,
+            "name": "Vaelden",
+            "gold": 42,
+            "experience": 43,
+            "guild": "God"
+        }
+    },
+    "skin": 1,
+    "current_pv": 100,
+    "position": {
+        "x": 0.1,
+        "y": 42.42,
+        "map": 0,
+        "instance": 2
+    },
+    "stats": {
+        "level": 1,
+        "strength": 2,
+        "dexterity": 3,
+        "constitution": 4,
+        "intelligence": 5,
+        "precision": 6,
+        "wisdom": 7
+    }
+},
+{
+    "id": 5,
+    "type": {
+        "monster": {
+            "id": 3,
+            "name": "zombie",
+            "behaviour_tree": "zombie.bt"
+        }
+    },
+    "skin": 0,
+    "current_pv": 100,
+    "position": {
+        "x": 0.0,
+        "y": 10.0,
+        "map": 0,
+        "instance": 3
+    },
+    "stats": {
+        "level": 7,
+        "strength": 6,
+        "dexterity": 5,
+        "constitution": 4,
+        "intelligence": 3,
+        "precision": 2,
+        "wisdom": 1
+    }
+}
+]
+```
+
+## POST /instances/:id/spawn
+
+Spawn a specific mob in the instance
+
+### Body
+
+```javascript
+{
+    "monster_class": 3,
+    "x": 0.0,
+    "y": 10.0
+}
+```
+
+Optional parameters:
+* `x`: Default value 0.0
+* `y`: Default value 0.0
+
+### Returns
+
+```javascript
+{
+    "id": 6,
+    "type": {
+        "monster": {
+            "id": 3,
+            "name": "zombie",
+            "behaviour_tree": "zombie.bt"
+        }
+    },
+    "current_pv": 100,
+    "position": {
+        "x": 0.0,
+        "y": 10.0,
+        "map": 0,
+        "instance": 3
+    },
+    "stats": {
+        "level": 7,
+        "strength": 6,
+        "dexterity": 5,
+        "constitution": 4,
+        "intelligence": 3,
+        "precision": 2,
+        "wisdom": 1
+    }
+}
+```
+
+## DELETE /instances/:id/entities/:id
+
+Removes a mob from the game
+
+XXX: Should it kick players as well?
+
+## GET /players
+
+Get list of connected players
+
+### Returns
+
+```javascript
+[
+{
+    "id": 4,
+    "type": {
+        "player": {
+            "id": 1,
+            "name": "Vaelden",
+            "gold": 42,
+            "experience": 43,
+            "guild": "God"
+        }
+    },
+    "skin": 1,
+    "current_pv": 100,
+    "position": {
+        "x": 0.1,
+        "y": 42.42,
+        "map": 0,
+        "instance": 2
+    },
+    "stats": {
+        "level": 1,
+        "strength": 2,
+        "dexterity": 3,
+        "constitution": 4,
+        "intelligence": 5,
+        "precision": 6,
+        "wisdom": 7
+    }
+}
+]
+```
+## DELETE /players/:id
+
+Kicks a player from the game
+
+XXX: Banning - Should we do it from the management console (i.e. block account, then kick from
+                                                            Lycan?)
+## POST /shutdown
+
```

Comment:
Like most APIs currently implemented: unit undefined and ignored :)

The main idea was to provide a way to shutdown / reboot the server easily. The timer could allow to send regular messages to all clients warning them of the downtime, but is not strictly required.

Also thinking more about it, the token (id) would only make sense if we want to schedule several reboots.

If we do, then I will add several routes `GET /shutdown` and `DELETE /shutdown` to manage everything.

In any case, those are in my opinion extremely low priority :)